crates/payload/basic/src/lib.rs

```diff
@@ -587,6 +603,73 @@ fn build_payload<Pool, Client>(
 
         let block_number = initialized_block_env.number.to::<u64>();
 
+        // Deposit transactions are always inserted at the top of the block built by the payload.
+        // These transactions are pre-paid on L1 and have no signature.
+        #[cfg(feature = "optimism")]
+        {
+            for deposit_tx in attributes.transactions {
+                // Check if the job was cancelled, if so we can exit early.
+                if cancel.is_cancelled() {
+                    return Ok(BuildOutcome::Cancelled)
+                }
+
+                // Convert the deposit transaction to a [TransactionSignedEcRecovered]. This is
+                // purely for the purposes of utilizing the [tx_env_with_recovered] function,
+                // deposit transactions do not have signatures.
+                let deposit_tx = deposit_tx
+                    .clone()
+                    .try_into_ecrecovered()
+                    .map_err(|_| PayloadBuilderError::DepositTransactionRecoverFailed)?;
+
+                // Configure the environment for the block.
+                let env = Env {
+                    cfg: initialized_cfg.clone(),
+                    block: initialized_block_env.clone(),
+                    tx: tx_env_with_recovered(&deposit_tx),
+                };
+
+                let mut evm = revm::EVM::with_env(env);
+                evm.database(&mut db);
+
+                let ResultAndState { result, state } = match evm.transact() {
+                    Ok(res) => res,
+                    Err(err) => {
+                        match err {
+                            EVMError::Transaction(err) => {
+                                // if the deposit transaction is invalid, we can skip it
+                                trace!(?err, ?deposit_tx, "skipping invalid deposit transaction");
+                                continue
+                            }
+                            err => {
+                                // this is an error that we should treat as fatal for this attempt
```

Comment:
I think we don't want to error here - failed deposits should still be included.

Reply:
at which step should deposit tx validation happen?

Reply:
Hm actually not sure if there is any explicit extra validation on the deposit txs. We check if the `gas_limit` payload attribute is present and that the transaction isn't malformed (in the first match arm), which falls back to the validation performed in revm's `Transact` trait impl. cc @protolambda